src/target/linux.rs

```diff
@@ -85,13 +94,76 @@ impl UnixTarget for LinuxTarget {
     fn pid(&self) -> Pid {
         self.pid
     }
+
+    fn unpause(&self) -> Result<WaitStatus, Box<dyn std::error::Error>> {
+        self.handle_breakpoint()?;
+        ptrace::cont(self.pid(), None)?;
+        let status = waitpid(self.pid(), None)?;
+
+        //// We may have hit a user defined breakpoint
+        if let WaitStatus::Stopped(_, nix::sys::signal::Signal::SIGTRAP) = status {
+            if let Some(bp) = self
+                .breakpoints
+                .borrow_mut()
+                .get_mut(&(self.read_regs()?.rip as usize - 1))
+            {
+                // Register which breakpoint was  hit
+                self.hit_breakpoint.borrow_mut().replace(bp.clone());
+
+                // Restore the program to its uninstrumented state
+                self.restore_breakpoint(bp)?;
+            };
+        }
+        Ok(status)
+    }
+
+    fn step(&self) -> Result<WaitStatus, Box<dyn std::error::Error>> {
+        if !self.handle_breakpoint()? {
+            ptrace::step(self.pid(), None).map_err(|e| Box::new(e))?;
+        }
+        let status = waitpid(self.pid(), None)?;
+        assert_eq!(
+            status,
+            WaitStatus::Stopped(self.pid(), nix::sys::signal::SIGTRAP)
+        );
```

Comment:
It could be a different signal, for example if an external process sent a signal racing with executing an instruction. This is most likely to happen when the instruction is a syscall.